app/src/main/java/io/appium/uiautomator2/handler/UpdateSettings.java

```diff
@@ -1,40 +1,54 @@
 package io.appium.uiautomator2.handler;
 
+import android.support.annotation.Nullable;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Map.Entry;
+
 import io.appium.uiautomator2.handler.request.SafeRequestHandler;
 import io.appium.uiautomator2.http.AppiumResponse;
 import io.appium.uiautomator2.http.IHttpRequest;
-import io.appium.uiautomator2.model.AllowInvisibleElements;
-import io.appium.uiautomator2.model.CompressedLayoutHierarchy;
 import io.appium.uiautomator2.model.Session;
-import io.appium.uiautomator2.model.WaitForIdleTimeout;
+import io.appium.uiautomator2.model.settings.AllowInvisibleElements;
+import io.appium.uiautomator2.model.settings.EnableNotificationListener;
+import io.appium.uiautomator2.model.settings.CompressedLayoutHierarchy;
+import io.appium.uiautomator2.model.settings.ISetting;
+import io.appium.uiautomator2.model.settings.WaitForIdleTimeout;
 import io.appium.uiautomator2.server.WDStatus;
 import io.appium.uiautomator2.utils.Logger;
-import java.util.Map;
 
 public class UpdateSettings extends SafeRequestHandler {
 
+    private static final Map<String, Class<? extends ISetting>> SETTINGS = new HashMap<String, Class<? extends ISetting>>() {
+        {
```

Comment:
I this inner block necessary?

Reply:
Yes - this is  [instance initializer block](https://docs.oracle.com/javase/tutorial/java/javaOO/initial.html)

Reply:
Ok. Thanks.